src/videostreamindex/VideoStreamIndex.ts

```diff
@@ -20,15 +20,21 @@ export default interface VideoStreamIndex {
   integrateIndexFrame(indexFrame: SdkIndexFrame): void;
 
   /**
-   * This function caches the currently ingested index frame (and may be eventually renamed). It can
-   * use the stored index as a backup for map functions like `groupIdForStreamId` and can be used for:
-   *   * Maintaining consistent information during a subscribe even if an index is received concurrently
-   *   * Maintaining information about previous senders simulcast streams to enable stream switching, e.g.
-   *     if a sender is transmitting low/hi and switches to just low, receivers need the previous index
-   *     to be able to know what group ID the high stream corresponds to.
+   * This function caches the currently ingested index frame for subscribe related activities (e.g. mapping tracks).
+   *
+   * The index will also use the stored index as a backup for map functions like `groupIdForStreamId`so that we can maintain information
+   * about previous senders simulcast streams to enable stream switching, e.g. if a sender is transmitting low/hi and
+   * switches to just low, receivers need the previous index to be able to know what group ID the high stream corresponds to.
    */
   subscribeFrameSent(): void;
 
+  /**
+   * This function caches the currently ingested index frame specifically as a backup for map functions like `groupIdForStreamId`so that we can maintain information
```

Comment:
Maybe I have not looked at video code for a while but both these methods sound very similar :D

Reply:
The issue is that two different sections of code want it to be cached at different times:
* On subscribe ack, we need the index when we subscribed, because it is the only one guaranteed to have the stream IDs used in the track mapping.
* On simulcast stream updates, we need the index of the last update, because that is the only one guranteed to have the stream IDs of the 'existing' subscription'
In a rewrite these probably should just be cached by what needs them (which is unfortunately the big context glob). But I just followed the existing pattern.